libraries/fc/include/fc/static_variant.hpp

```diff
@@ -14,587 +14,86 @@
-        }
-    }
-    template<typename visitor>
-    auto visit(visitor&& v) & {
-        return impl::storage_ops_auto<visitor, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-
-    template<typename visitor>
-    auto visit(visitor&& v) const & {
-        return impl::storage_ops_const_auto<visitor, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-    template<typename visitor>
-    auto visit(visitor&& v) && {
-        return impl::storage_ops_revalue_ref_auto<visitor, 0, Types...>::apply_rvalue_ref(_tag, storage, std::forward<visitor>(v));
-    }
-
-    template<typename R, typename visitor>
-    auto visit(visitor&& v) & {
-        return impl::storage_ops<R, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-
-    template<typename R, typename visitor>
-    auto visit(visitor&& v) const & {
-        return impl::storage_ops<R, 0, Types...>::apply(_tag, storage, std::forward<visitor>(v));
-    }
-
-    template<typename R, typename visitor>
-    auto visit(visitor&& v) && {
-        return impl::storage_ops<R, 0, Types...>::apply_rvalue_ref(_tag, storage, std::forward<visitor>(v));
-    }
-
-    static uint32_t count() { return type_info::count; }
-    void set_which( uint32_t w ) {
-      FC_ASSERT( w < count()  );
-      this->~static_variant();
-      try {
-         _tag = w;
-         impl::storage_ops<void, 0, Types...>::con(_tag, storage);
-      } catch ( ... ) {
-         _tag = 0;
-         impl::storage_ops<void, 0, Types...>::con(_tag, storage);
-      }
-    }
-
-    int which() const {return _tag;}
-
-    template<typename X>
-    bool contains() const { return which() == tag<X>::value; }
-
-    template<typename X>
-    static constexpr int position() { return impl::position<X, Types...>::pos; }
-
-    template<int Pos, std::enable_if_t<Pos < type_info::size,int> = 1>
-    using type_at = typename impl::type_at<Pos, Types...>::type;
-};
-
-template<typename Result>
-struct visitor {
+  template<typename T> void operator()( T& v )const
+  {
+      from_variant( var, v );
+  }
 };
 
-   struct from_static_variant
-   {
-      variant& var;
-      from_static_variant( variant& dv ):var(dv){}
-
-      template<typename T> void operator()( const T& v )const
-      {
-         to_variant( v, var );
-      }
-   };
-
-   struct to_static_variant
-   {
-      const variant& var;
-      to_static_variant( const variant& dv ):var(dv){}
-
-      template<typename T> void operator()( T& v )const
-      {
-         from_variant( var, v );
-      }
-   };
-
+template<typename... T> void to_variant( const std::variant<T...>& s, fc::variant& v )
+{
+  variants vars(2);
+  vars[0] = s.index();
+  visit( from_static_variant(vars[1]), s );
+  v = std::move(vars);
+}
+template<typename... T> void from_variant( const fc::variant& v, std::variant<T...>& s )
+{
+  auto ar = v.get_array();
+  if( ar.size() < 2 ) return;
```

Comment:
Does s need to be void (initialized) if size() < 2, or caller needs to initialize?

Reply:
This is actually pre-existing functionality that was needed after removing `fc::static_variant`.  I'm not quite sure of if that is the intent or not.

Reply:
Seems like converting to std::variant, if it is empty, can we set it as std::monostate.

Reply:
We can only set it to the monostate if that type is part of the variant's type set and this is done by default constructing a variant instance.  So I will default construct a variant instance and assign it to `s`.